From 967ba6898a3e3a29070b209d0fbf71aa580085d6 Mon Sep 17 00:00:00 2001
From: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Date: Wed, 20 Jul 2022 19:40:55 +0300
Subject: [PATCH 044/102] hse: add shared secret key group to ram catalog

A shared secret key slot must be added to the RAM Catalog so that HSE's
driver in OP-TEE can use it for key derivation. However, since the RAM
Key slots are limited to only 20 slots, reduce the number of AES key
slots by 1.

Issue: ALB-8819
Upstream-Status: Pending 

Signed-off-by: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/s32-cc/hse_adv_secboot.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c b/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
index e4abe18ebe..a4d1562d53 100644
--- a/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
+++ b/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
@@ -44,11 +44,13 @@ DECLARE_GLOBAL_DATA_PTR;
 {HSE_ALL_MU_MASK, HSE_KEY_OWNER_ANY, \
 	HSE_KEY_TYPE_RSA_PUB, 1u, HSE_KEY2048_BITS }, \
 {HSE_ALL_MU_MASK, HSE_KEY_OWNER_ANY, \
-	HSE_KEY_TYPE_AES, 12u, HSE_KEY256_BITS }, \
+	HSE_KEY_TYPE_AES, 11u, HSE_KEY256_BITS }, \
 {HSE_ALL_MU_MASK, HSE_KEY_OWNER_ANY, \
 	HSE_KEY_TYPE_HMAC, 6u, HSE_KEY512_BITS}, \
 {HSE_ALL_MU_MASK, HSE_KEY_OWNER_ANY, \
 	HSE_KEY_TYPE_ECC_PUB, 1u, HSE_KEY256_BITS}, \
+{HSE_ALL_MU_MASK, HSE_KEY_OWNER_ANY, \
+	HSE_KEY_TYPE_SHARED_SECRET, 1u, HSE_KEY256_BITS}, \
 {0u, 0u, 0u, 0u, 0u}
 
 #define APP_CODE_OFFSET 0x40
-- 
2.17.1

